substrate/frame/support/src/traits/tokens/fungible/union_of.rs

```diff
@@ -0,0 +1,1005 @@
+			Left(()) => <Left as fungible::Unbalanced<AccountId>>::set_total_issuance(amount),
+			Right(a) => <Right as fungibles::Unbalanced<AccountId>>::set_total_issuance(a, amount),
+		}
+	}
+	fn decrease_balance(
+		asset: Self::AssetId,
+		who: &AccountId,
+		amount: Self::Balance,
+		precision: Precision,
+		preservation: Preservation,
+		force: Fortitude,
+	) -> Result<Self::Balance, DispatchError> {
+		match Criterion::convert(asset) {
+			Left(()) => <Left as fungible::Unbalanced<AccountId>>::decrease_balance(
+				who,
+				amount,
+				precision,
+				preservation,
+				force,
+			),
+			Right(a) => <Right as fungibles::Unbalanced<AccountId>>::decrease_balance(
+				a,
+				who,
+				amount,
+				precision,
+				preservation,
+				force,
+			),
+		}
+	}
+	fn increase_balance(
+		asset: Self::AssetId,
+		who: &AccountId,
+		amount: Self::Balance,
+		precision: Precision,
+	) -> Result<Self::Balance, DispatchError> {
+		match Criterion::convert(asset) {
+			Left(()) =>
+				<Left as fungible::Unbalanced<AccountId>>::increase_balance(who, amount, precision),
+			Right(a) => <Right as fungibles::Unbalanced<AccountId>>::increase_balance(
+				a, who, amount, precision,
+			),
+		}
+	}
+}
+
+impl<
+		Left: fungible::UnbalancedHold<AccountId>,
+		Right: fungibles::UnbalancedHold<AccountId, Balance = Left::Balance, Reason = Left::Reason>,
+		Criterion: Convert<AssetKind, Either<(), Right::AssetId>>,
+		AssetKind: AssetId,
+		AccountId,
+	> fungibles::UnbalancedHold<AccountId> for UnionOf<Left, Right, Criterion, AssetKind, AccountId>
+{
+	fn set_balance_on_hold(
+		asset: Self::AssetId,
+		reason: &Self::Reason,
+		who: &AccountId,
+		amount: Self::Balance,
+	) -> DispatchResult {
+		match Criterion::convert(asset) {
+			Left(()) => <Left as fungible::UnbalancedHold<AccountId>>::set_balance_on_hold(
+				reason, who, amount,
+			),
+			Right(a) => <Right as fungibles::UnbalancedHold<AccountId>>::set_balance_on_hold(
+				a, reason, who, amount,
+			),
+		}
+	}
+	fn decrease_balance_on_hold(
+		asset: Self::AssetId,
+		reason: &Self::Reason,
+		who: &AccountId,
+		amount: Self::Balance,
+		precision: Precision,
+	) -> Result<Self::Balance, DispatchError> {
+		match Criterion::convert(asset) {
+			Left(()) => <Left as fungible::UnbalancedHold<AccountId>>::decrease_balance_on_hold(
+				reason, who, amount, precision,
+			),
+			Right(a) => <Right as fungibles::UnbalancedHold<AccountId>>::decrease_balance_on_hold(
+				a, reason, who, amount, precision,
+			),
+		}
+	}
+	fn increase_balance_on_hold(
+		asset: Self::AssetId,
+		reason: &Self::Reason,
+		who: &AccountId,
+		amount: Self::Balance,
+		precision: Precision,
+	) -> Result<Self::Balance, DispatchError> {
+		match Criterion::convert(asset) {
+			Left(()) => <Left as fungible::UnbalancedHold<AccountId>>::decrease_balance_on_hold(
+				reason, who, amount, precision,
+			),
+			Right(a) => <Right as fungibles::UnbalancedHold<AccountId>>::decrease_balance_on_hold(
```

Comment:
should probably add tests for all these methods to catch regressions or other small typos 

```suggestion
			Left(()) => <Left as fungible::UnbalancedHold<AccountId>>::increase_balance_on_hold(
				reason, who, amount, precision,
			),
			Right(a) => <Right as fungibles::UnbalancedHold<AccountId>>::increase_balance_on_hold(
```